### PR TITLE
Remove unneeded dependencies, sort dependencies alphabetically

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@jpkrohling could you please review and eventually merge?

File: hawkular-bus/hawkular-bus-common/pom.xml [hunk 1]
```diff
@@ -37,6 +37,18 @@
   </properties>
 
   <dependencies>
+    <!-- To avoid javadoc errors (and thus causing mvn release plugin to fail) we need this provided dependency.
```
Comment (jpkrohling) [inline hawkular-bus/hawkular-bus-common/pom.xml]: Is this really needed? I've seen similar comments on the `hawkular/modules`, and the javadoc passed/failed regardless of the dependency.

File: hawkular-bus/pom.xml [hunk 2]
```diff
@@ -79,28 +69,11 @@
...
       <dependency>
-        <groupId>org.wildfly.core</groupId>
-        <artifactId>wildfly-controller</artifactId>
-        <version>${version.org.wildfly.core}</version>
-      </dependency>
-
-      <dependency>
         <groupId>org.wildfly</groupId>
-        <artifactId>wildfly-dist</artifactId>
+        <artifactId>wildfly-jms-client-bom</artifactId>
```
Comment (jpkrohling) [inline hawkular-bus/pom.xml]: I'm sure this dependency is correct, but just to confirm: isn't this also included on the main wildfly-javaee bom?

**Comment by jpkrohling** [comment]
Review done.

File: hawkular-bus/hawkular-bus-common/pom.xml (hunk 1 above)
Comment (ppalaga) [inline hawkular-bus/hawkular-bus-common/pom.xml]: > Is this really needed? I've seen similar comments on the hawkular/modules, and the javadoc passed/failed regardless of the dependency.

No idea. This comes from you @jmazzitelli ? Can it eventually be removed?

File: hawkular-bus/pom.xml (hunk 2 above)
Comment (ppalaga) [inline hawkular-bus/pom.xml]: Clicking through "usages" on http://mvnrepository.com/artifact/org.wildfly/wildfly-jms-client-bom/10.0.0.Final, it does not look like it was linked with wildfly-javaee bom. Neither I can see it anywhere when studying wildfly-jms-client-bom.

**Pull request merged**